src/profile/ProfileForm.tsx

```diff
@@ -37,585 +25,385 @@ interface ChangedUser {
           </Col>
         </Form.Group>
-      )}
-
-      <Form.Group as={Row} controlId="profileNewPassword">
-        <Form.Label column sm={2}>
-          New password
-          <OverlayTrigger placement="right" overlay={password_info}>
-            <span>(i)</span>
-          </OverlayTrigger>
-        </Form.Label>
-        <Col sm={10}>
-          <Form.Row>
-            <Col sm="auto">
-              <Button
-                variant="secondary"
-                onClick={() => {
-                  dispatch({
-                    type: "set data",
-                    data: { password: nanoid() },
-                    errors: { status_code: 0, password: [] },
-                  });
-                  setShowPassword(true);
+        <Form.Group as={Row} controlId="writerName">
+          <Form.Label column sm={2}>
+            Writer Name
+          </Form.Label>
+          <Col sm={10}>
+            <Field
+              errors={props.context.errors}
+              type="text"
+              name="writer_name"
+              ref={props.context.register({
+                required: "USER.REQUIRED.writer_name",
+              })}
+            />
+          </Col>
+        </Form.Group>
+        {auth.isEditor() && (
+          <Form.Group as={Row} controlId="isEditor">
+            <Form.Label column sm={2}>
+              Role
+            </Form.Label>
+            <Col sm={10}>
+              <Form.Control
+                as="select"
+                onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
+                  if (e.currentTarget.value === "editor") {
+                    props.context.setValue("is_editor", true);
+                  } else if (e.currentTarget.value === "contributor") {
+                    props.context.setValue("is_editor", false);
+                  }
                 }}
+                value={
+                  props.context.getValues().is_editor ? "editor" : "contributor"
+                }
+                className="custom-select"
+                custom
               >
-                Generate
-              </Button>
+                <option value="contributor">Contributor</option>
+                <option value="editor">Editor</option>
+              </Form.Control>
             </Col>
-            <Col>
-              <InputGroup>
-                <Form.Control
+          </Form.Group>
+        )}
+        <Form.Group as={Form.Row} controlId="password">
+          <Form.Label column sm={2}>
+            New Password
+          </Form.Label>
+          <Col sm={10}>
+            <Form.Row>
+              <Col sm="auto">
+                <Button
+                  className="w-100"
+                  variant="secondary"
+                  onClick={() => {
+                    props.context.setValue("password", nanoid());
+                    // set the password visible so you can see what you get
+                    setShowPassword(true);
+                  }}
```

Comment:
When you generate a password, it doesn't trigger revalidation of the field; you have to change focus, then the onBlur validation kicks in: https://streamable.com/53ngus

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/13/commits/07fb0662c68d0190ccfafe88acfde9fc6d2d29bf